lib/userFactory.js

```diff
@@ -65,6 +65,38 @@ class UserFactory extends BaseFactory {
     }
 
     /**
+     * Get a user model by name, id, or token
+     * @method get
+     * @param   {Mixed}     config
+     * @param   {String}    [config.token]      Token of the user to look up
+     * @return  {Promise}
+     */
+    get(config) {
+        if (!config || !config.token) {
+            return super.get(config);
+        }
+
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const TokenFactory = require('./tokenFactory');
+        /* eslint-enable global-require */
+        const tokenFactory = TokenFactory.getInstance();
+
+        return tokenFactory.get({ value: config.token })
```

Comment:
Would this work? From `baseFactory`, it looks like you can only search by fields that are in `keys`: https://github.com/screwdriver-cd/models/blob/master/lib/baseFactory.js#L73-L77

And `value` is not in token's keys: https://github.com/screwdriver-cd/data-schema/blob/master/models/token.js#L102